Include/Rover/Model.hpp

```diff
@@ -122,15 +123,15 @@ namespace Rover {
     auto generator = std::mt19937(std::random_device()());
     std::iota(order.begin(), order.end(), std::size_t(0));
     auto basis = trial[0];
-    visit_arguments([&](auto& result, auto i) {
+    ArgumentVisitor<Arguments>::visit([&](auto& result, auto i) {
```

Comment:
Can you go back to having a ```visit_arguments``` function that is implemented by calling ```ArgumentVisitor<Arguments>::visit```? I'd prefer concealing ```ArgumentVisitor``` behind a function.